controllers/hetznerbaremetalhost_controller.go

```diff
@@ -180,19 +180,26 @@ func (r *HetznerBareMetalHostReconciler) reconcileSelectedStates(ctx context.Con
 	switch bmHost.Spec.Status.ProvisioningState {
 	// Handle StateNone: check whether needs to be provisioned or deleted.
 	case infrav1.StateNone:
-		var needsUpdate bool
-		if !bmHost.DeletionTimestamp.IsZero() && bmHost.Spec.ConsumerRef == nil {
-			bmHost.Spec.Status.ProvisioningState = infrav1.StateDeleting
-			needsUpdate = true
+		if !bmHost.DeletionTimestamp.IsZero() {
+			if !conditions.Has(bmHost, infrav1.HostReadyCondition) {
```

Comment:
you shouldn't do this check here. You should set it independently of whether it exists or not. You even have to overwrite it. 

What I meant earlier was that you might still keep needsUpdate and then need to check whether the condition you set actually changed something in the object or not. But for that you need to check the reason as well, not only that the condition exists. 

I would just remove this if statement and always set the condition on false and you are good to go!